### ISSUE TITLE
Request-URI Too Long

### ISSUE THREAD

**Issue by WKleinschmit** [JetBrains/YouTrackSharp#94]
Thr following code:

```c#
BearerTokenConnection ytConnection = new BearerTokenConnection(
 "https://youtrack.XXX.com/",
 "perm:...");

Issue issue = new Issue
{
    Description = description.ToString(),
    Summary = "Summary",
};

IIssuesService issuesService = ytConnection.CreateIssuesService();

string id = await issuesService.CreateIssue("XXX", issue);
```
Results in a System.Net.Http.HttpRequestException with the message
"Response status code does not indicate success: 414 (Request-URI Too Long)."

What am I doing wrong here?

**Comment by maartenba** [comment]
Think that mostly depends on the length of the description and summary in the issue, could you try a shorter one?

**Comment by WKleinschmit** [comment]
I can try. But I need the (admittadly long) description. Is there a way to edit it after creation of the issue?

Also: Shouldn't the description be transmitted in the body of the request as stated here: https://www.jetbrains.com/help/youtrack/standalone/2019.2/api-howto-create-issue.html

**Comment by maartenba** [comment]
That's a newer API, will see if I can push a change for that.

**Comment by WKleinschmit** [comment]
Can't wait...

**Comment by Jaben** [comment]
The latest release has been updated to YouTrack 2020 client -- please try it now.

**Comment by Jaben** [comment]
@maartenba is there an ETA for to move to the newer REST API? Is the workaround to truncate the Summary or Description or both? If so, what is the max length?

Thanks.

**Comment by maartenba** [comment]
@Jaben There is no ETA right now.

As for max length, this should be ~10000 characters.

**Comment by maartenba** [comment]
@Jaben Regarding new API, if needed you can use the OpenAPI spec (https://www.jetbrains.com/help/youtrack/standalone/openapi-specification.html) and generate a client (e.g. using [NSwagStudio](https://github.com/RicoSuter/NSwag/wiki/NSwagStudio)). This may be the route we will take as well (with helpers for many methods)

**Comment by rekolobov** [comment]
Fixed by #105 and 2021.3.0 release
